src/mlpack/tests/decision_tree_test.cpp

```diff
@@ -471,6 +471,9 @@ TEST_CASE("RandomBinaryNumericSplitNoGainTest", "[DecisionTreeTest]")
         bestGain, values, labels, 2, weights, 3, 1e-7, classProbabilities1,
         aux1);
 
+    // The variable is not used; suppress warnings.
+    (void) gain;
```

Comment:
Hm, I guess that should work, any reason to keep `gain`, looks like it's not used inside the for loop.

Reply:
You're right, I switched to a solution that might be easier.  Also I fixed a bunch of warnings while I was at it (at least they show up with gcc 10.2). :)